Assets/MixedRealityToolkit.SDK/Experimental/ScrollingObjectCollection/ScrollingObjectCollection.cs

```diff
@@ -0,0 +1,2232 @@
+        /// </summary>
+        public float CellWidth
+        {
+            get { return (cellWidth > 0) ? cellWidth : 0.00001f; }
+            set { cellWidth = value; }
+        }
+
+        [Tooltip("Height of cell per object.")]
+        [SerializeField]
+        [Range(0.00001f, 100.0f)]
+        private float cellHeight = 0.25f;
+
+        /// <summary>
+        /// Height of the cell per object in the collection.
+        /// </summary>
+        public float CellHeight
+        {
+            get { return (cellHeight > 0) ? cellHeight : 0.00001f; }
+            set { cellHeight = value; }
+        }
+
+        [SerializeField]
+        [Tooltip("Multiplier to add more bounce to the overscroll of a list when using VelocityType.FalloffPerFrame or VelocityType.FalloffPerItem.")]
+        private float bounceMultiplier = 0.1f;
+
+        /// <summary>
+        /// Multiplier to add more bounce to the overscroll of a list when using <see cref="VelocityType.FalloffPerFrame"/> or <see cref="VelocityType.FalloffPerItem"/>.
+        /// </summary>
+        public float BounceMultiplier
+        {
+            get { return bounceMultiplier; }
+            set { bounceMultiplier = value; }
+        }
+
+        /// <summary>
+        /// The UnityEvent type the ScrollingObjectCollection sends.
+        /// GameObject is the object the fired the scroll.
+        /// </summary>
+        [System.Serializable]
+        public class ScrollEvent : UnityEvent<GameObject> { }
+
+        /// <summary>
+        /// Event that is fired on the target object when the ScrollingObjectCollection deems event as a Click.
+        /// </summary>
+        [Tooltip("Event that is fired on the target object when the ScrollingObjectCollection deems event as a Click.")]
+        public ScrollEvent ClickEvent = new ScrollEvent();
+
+        /// <summary>
+        /// Event that is fired on the target object when the ScrollingObjectCollection is touched.
+        /// </summary>
+        [Tooltip("Event that is fired on the target object when the ScrollingObjectCollection is touched.")]
+        public ScrollEvent TouchStarted = new ScrollEvent();
+
+        /// <summary>
+        /// Event that is fired on the target object when the ScrollingObjectCollection is no longer touched.
+        /// </summary>
+        [Tooltip("Event that is fired on the target object when the ScrollingObjectCollection is no longer touched.")]
+        public ScrollEvent TouchEnded = new ScrollEvent();
+
+        /// <summary>
+        /// Event that is fired on the target object when the ScrollingObjectCollection is no longer in motion from velocity
+        /// </summary>
+        [Tooltip("Event that is fired on the target object when the ScrollingObjectCollection is no longer in motion from velocity.")]
+        public UnityEvent ListMomentumEnded = new UnityEvent();
+
+        /// <summary>
+        /// First item (visible) in the <see cref="ViewableArea"/>. 
```

Comment:
Update this to how would I use this public field. FirstItemInViewIndex for NodeList etc